frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.unit.spec.tsx

```diff
@@ -1 +1,106 @@
-// TODO
+import userEvent from "@testing-library/user-event";
+import type { DashboardOrderedTab } from "metabase-types/api";
+import {
+  createMockDashboard,
+  createMockDashboardOrderedCard,
+} from "metabase-types/api/mocks";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { getDefaultTab } from "metabase/dashboard/actions";
+import { renderWithProviders, screen, waitFor, within } from "__support__/ui";
+import { DashCardTabMenu } from "./DashCardTabMenu";
+
+const DASHCARD = createMockDashboardOrderedCard();
+
+const TEST_DASHBOARD = createMockDashboard({
+  ordered_cards: [DASHCARD],
+});
+
+const TAB_1 = getDefaultTab({ tabId: 1, dashId: 1, name: "Tab 1" });
+const TAB_2 = getDefaultTab({ tabId: 2, dashId: 1, name: "Tab 2" });
+const TAB_3 = getDefaultTab({ tabId: 3, dashId: 1, name: "Tab 3" });
+
+const setup = ({
+  tabs = [TAB_1, TAB_2, TAB_3],
+  selectedTab = TAB_1,
+}: {
+  tabs?: DashboardOrderedTab[];
+  selectedTab?: DashboardOrderedTab;
+}) => {
+  const dashboard = createMockDashboardState({
+    dashboardId: TEST_DASHBOARD.id,
+    selectedTabId: selectedTab.id,
+    dashboards: {
+      [TEST_DASHBOARD.id]: {
+        ...TEST_DASHBOARD,
+        ordered_cards: TEST_DASHBOARD.ordered_cards.map(c => c.id),
+        ordered_tabs: tabs,
+      },
+    },
+  });
+
+  return renderWithProviders(<DashCardTabMenu dashCardId={123} />, {
+    storeInitialState: createMockState({ dashboard }),
+  });
+};
+
```

Comment:
all the tests needs to be re-done as the feature changed
I'll rewrite them after the testing plan has been reviewed